### PR TITLE
feat: Add a simple polyfill system

### PR DESCRIPTION
**Pull request by bhelx**
This adds a basic npm project which will be imported into core to polyfill globals that lots of JS programs expect.

```
$ extism call plugin.wasm evalcode --input=(new TextEncoder()).encode('€') --wasi
# => 226,130,172
```

Closes https://github.com/extism/js-pdk/issues/27

### CODE REVIEW

**Review by zshipko — APPROVED** [review]

**Comment by nilslice** [comment]
excellent!

**Comment by bhelx** [comment]
Going to double check the size difference this creates. If it's too big i'm going to add a flag so that you have to opt into including the polyfills

**Comment by bhelx** [comment]
This only partly works. Doesn't seem to be able to find TextEncoder when included as a dependency. need to do a little more work on this. Might be a good time to see if we can support ESM as i suspect that might help.

**Comment by bhelx** [comment]
This is on hold but I'm going to come back to it after some refactors. Although it's nice to do all of this in JS, i think some of these should be in rust.

File: bundle.js [hunk 1]
```diff
@@ -25,6 +35,7 @@ __export(script_exports, {
   i_error_out: () => i_error_out
 });
 module.exports = __toCommonJS(script_exports);
+var import_react = __toESM(require("react"));
```
Comment (nilslice) [inline bundle.js:38]: where does this come from? are we providing react to all PDK code?

Comment (bhelx) [reply bundle.js:38]: my bad, this accidentally got added with some testing i was doing i think. I moved this PR back to draft as i'm re-working it along with getting react to work.

Comment (bhelx) [reply bundle.js:38]: also this is a test file i'm going to be deleting soon. there is some testing stuff on the top level

Comment (nilslice) [reply bundle.js:38]: Ok gotcha! just saw it was marked ready and was giving it a quick look -- but will keep eyes elsewhere while your drafting :smile:

**Comment by bhelx** [comment]
I think this is in a better spot now. Uses rust to do text encoding and decoding and it's just exposed in JS.

```
$ extism call plugin.wasm evalcode --wasi --input="new TextEncoder().encode('über')"
# => 195,188,98,101,114
```

**Comment by bhelx** [comment]
Also i've gotten it work work with react so we know it seems to work in the real world

**Comment by bhelx** [comment]
Will have some follow ups here but good enough to move forward

**Pull request merged**